lang/syn/src/codegen/accounts/bumps.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{
+    codegen::accounts::{generics, ParsedGenerics},
+    *,
+};
+use std::fmt::Display;
+
+use super::constraints;
+
+pub fn generate_bumps_name<T: Display>(anchor_ident: &T) -> Ident {
+    Ident::new(&format!("{}Bumps", anchor_ident), Span::call_site())
+}
+
+pub fn generate(accs: &AccountsStruct) -> proc_macro2::TokenStream {
+    let name = &accs.ident;
+    let bumps_name = generate_bumps_name(name);
+    let ParsedGenerics {
+        combined_generics,
+        trait_generics: _,
+        struct_generics,
+        where_clause,
+    } = generics(accs);
+
+    let bump_fields: Vec<proc_macro2::TokenStream> = accs
+        .fields
+        .iter()
+        .filter_map(|af| {
+            let ident = af.ident();
+
+            match af {
+                AccountField::Field(f) => {
+                    let constraints = constraints::linearize(&f.constraints);
+                    let bump_field = quote!(pub #ident: u8);
+
+                    for c in constraints.iter() {
+                        // Verify this in super::constraints
+                        // The bump is only cached if
+                        // - PDA is marked as init
+                        // - PDA is not init, but marked with bump without a target
```

Comment:
Just to confirm, does this intentionally not cache when the `bump` is explicit, i.e. `bump = target`?

Reply:
Yep, we don't need to get the bump if we passed it with account or arguments